server.go

```diff
@@ -34,15 +36,82 @@ func (s *Server) Register(namespace string, service Invoker) {
 }
 
 // process process JSON-RPC 2.0 message, invokes correct method for namespace and returns JSON-RPC 2.0 Response.
-func (s *Server) process(ctx context.Context, message json.RawMessage) Response {
-	req := &Request{}
+func (s *Server) process(ctx context.Context, message json.RawMessage) interface{} {
 
-	// unmarshal request
-	// TODO process batch
-	if err := json.Unmarshal(message, req); err != nil {
+	requests := []Request{}
+
+	// parsing batch requests
+	batch := false
+	for _, b := range message {
+		if unicode.IsSpace(rune(b)) {
+			continue
+		}
+
+		if b == '[' {
+			batch = true
+		}
+		break
+	}
+
+	// making not batch request looks like batch to simplify further code
+	if !batch {
+		message = append(append([]byte{'['}, message...), ']')
+	}
+
+	// unmarshal request(s)
+	if err := json.Unmarshal(message, &requests); err != nil {
 		return NewResponseError(nil, ParseError, "", nil)
 	}
 
+	// if there no requests to process
+	if len(requests) == 0 {
+		return NewResponseError(nil, InvalidRequest, "", nil)
+	}
+
+	// running request asynchronously
+	reqLen := len(requests)
+	respChan := make(chan Response, reqLen)
+	var wg sync.WaitGroup
+	wg.Add(reqLen)
```

Comment:
i think that we need to limit max running parallel requests, parameter should be public Server variable  with default value `10` set in NewServer(). If reqLen > MaxBatchRequests then return InvalidRequest with correct message.